src/OpenTelemetry/Logs/ILogger/OpenTelemetryLoggingExtensions.cs

```diff
@@ -173,6 +173,12 @@ public static class OpenTelemetryLoggingExtensions
         // Note: This will bind logger options element (e.g., "Logging:OpenTelemetry") to OpenTelemetryLoggerOptions
         RegisterLoggerProviderOptions(services);
 
+        // Note: We disable built-in IOptionsMonitor features for
+        // OpenTelemetryLoggerOptions as a workaround to prevent unwanted
+        // objects (processors, exporters, etc.) being created by
+        // configuration delegates during reload of IConfiguration.
+        services.DisableOptionsMonitor<OpenTelemetryLoggerOptions>();
```

Comment:
This PR is changing OTel to use `IOptions` (instead of `IOptionsMonitor`) for `OpenTelemetryLoggerOptions`. But my thinking is, we can't prevent users from accessing `IOptionsMonitor` and we can't prevent some future dev from re-introducing it. The infrastructure here is so we can make it deterministic and have unit tests validating it will work correctly should `IOptionsMonitor<OpenTelemetryLoggerOptions>` sneak into the processor anywhere.

Reply:
1. Do we really expect someone outside of OpenTelemetry to try to get an `IOptionsMonitor<OpenTelemetryLoggerOptions>`? What would they do with it?
2. If someone does explicitly use `IOptionsMonitor`, don't they want updates as the config changes? That's why they choose IOptionsMonitor.

Would it be better to just fail in this case if we explicitly want to block it?

Reply:
While it's unclear why anyone else would need to monitor `OpenTelemetryLoggerOptions` for changes when the core library won't react to changes itself, but it feels like code smell to go out of our way prevent it by replacing core options services and changing how they work. What if someone wants to just new up an `OpenTelemetryLoggerOptions` and calls `options.AddProcessor(new BatchLogRecordExportProcessor(new OtlpLogExporter(new()))` in a test or something? Should they know that this will spawn a thread that will never get stopped?

I think the core issue is how `BatchExportProcessor<T>` spawns a thread in its constructor.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/876e4faa5aac56f456b52e4fbca965ce0e7b9b13/src/OpenTelemetry/BatchExportProcessor.cs#L60-L65

I know that this API has already shipped, so adding a `StartAsync` method or something like that may not be feasible, but could you unseal `OnStart` in the base class and start the thread there? Or lazily start the thread some other way? If not, should we deprecate the `BaseProcessor<LogRecord>` overload of `AddProcessor()` and tell people to use the `Func<IServiceProvider, BaseProcessor<LogRecord>` overload instead?

If we have to override the `IOptionsMonitor<OpenTelemetryLoggerOptions>`, I agree with @eerhardt that it should throw from everything with a `NotSupportedException` indicating that reloading `OpenTelemetryLoggerOptions` is completely unsupported.

Reply:
I just pushed updates so that accessing `OpenTelemetryLoggerOptions` via `IOptionsMonitor` or `IOptionsSnapshot` will result in a `NotSupportedException`. Technically breaking, but I think the impact will be very low.

> If not, should we deprecate the BaseProcessor<LogRecord> overload of AddProcessor() and tell people to use the Func<IServiceProvider, BaseProcessor<LogRecord> overload instead?

This is the plan yes (more or less)! We have another API for building logging pipelines which does not suffer from these issues. The plan is to make that a stable API (#5442) and then we can `Obsolete` these `AddProcessor` methods on `OpenTelemetryLoggerOptions`.

Reply:
Good discussion and thanks for the suggestions/ideas!

While I agree with most parts here, I suggest that we focus on "mitigate the issue quickly with minimum change/risk" instead of trying to get a complete solution in this PR. I'm specifically concerned about throwing exception in this PR. I suggest that we take the feedback and think about exception or other approaches in a follow up PR once the mitigation/hotfix is released.

Reply:
> I suggest that we focus on "mitigate the issue quickly with minimum change/risk" instead of trying to get a complete solution in this PR

So that means just changing our references of `IOptionsMonitor<OpenTelemetryLoggerOptions>` => `IOptions<OpenTelemetryLoggerOptions>`? Don't do anything to prevent users from accessing IOptionsMonitor. Am I understanding that correctly?

If so, I agree that since this is a hotfix, keep it scoped to resolving the issue at hand.

Reply:
I don't have strong opinion regarding the actual solution (e.g. IOptions/IOptionsMonitor, singleton, etc.), as long as it meets the two conditions https://github.com/open-telemetry/opentelemetry-dotnet/pull/5514#pullrequestreview-1995577792 I'm good 👍.